yt/frontends/boxlib/data_structures.py

```diff
@@ -566,6 +566,11 @@ def _read_particles(self, directory_name, is_checkpoint, extra_field_names=None)
                                                         is_checkpoint,
                                                         extra_field_names)
 
+        num_parts = self.particle_headers[directory_name].num_particles
+        if not hasattr(self.ds, "_particle_type_counts") or self.ds._particle_type_counts is None:
```

Comment:
I don’t think there’s any harm in unconditionally clobbering this with an empty dict.

Reply:
There can be multiple entries - this function will be called once for every particle type.

Reply:
Then I'm a little confused why you need to check if the attribute exists at all. It looks like it will always be initialized to `None`:

https://github.com/yt-project/yt/blob/c122c97bddf36605f37a2eb122c7fb43c903b464/yt/data_objects/static_output.py#L243